index.js

```diff
@@ -11,6 +16,52 @@ module.exports = {
     this.registerModels(customModels);
   },
 
+  async initializeMoneyMaker(currency) {
+    currency ??= 'XXX';
+
+    // to be called after db initialization
+    let values;
+    try {
+      // error thrown if migration hasn't taken place
+      values = await frappe.db.getSingleValues(
+        {
+          fieldname: 'internalPrecision',
+          parent: 'SystemSettings',
+        },
+        {
+          fieldname: 'displayPrecision',
+          parent: 'SystemSettings',
+        }
+      );
+    } catch {
```

Comment:
if we know exactly what error class/type is thrown when migration has not happened we can catch that and avoid catching general error cases?